### DISCUSSION TITLE
How to perform global exception capture for Route Handlers

### DISCUSSION THREAD

**Discussion by ljianc** [vercel/next.js#59868] [App Router] [answered]
At first, I tried using `middleware`, but found that it only applies to  before the request start，It cannot capture exceptions within every Route Handler。
How to capture unhandled exceptions globally instead of writing duplicate `try catch` code for each Route Handler

**Comment by ananuness** [comment, marked as answer]
You can create a helper function for that and wrap your route handlers with that:

1. Create the error handler:
```js
import { NextRequest, NextResponse } from "next/server";

function errorHandler(fn) {
  return async function (request, ...args) {
    try {
      return await fn(request, ...args);
    } catch (error) {
      return NextResponse.json(
        { message: error.message },
        { status: 500 }
      );
    }
  };
}

export default errorHandler;
```

2. Wrap your route handler:
```js
export const GET = errorHandler(async (request) => {
  // your logic here

  return NextResponse.json({ message: "No errors" }, { status: 200 });
});
```
I hope it helps you!

**Reply by ljianc** [reply]
At the beginning, follow Java and ASP.NET way of thinking, I'm trying to find a global entry point to handle this. After reading your reply, I only renamed `errorHandler` to `$`, It doesn't seem to increase duplicate code. This is a very simple and perfect solution. Thank you very much

**Reply by ananuness** [reply]
Yees! I tried it too! But it doesn't seem to have it (yet). This workaround helps a lot! I'm glad it helped you!
